internal/agent/device/config/config.go

```diff
@@ -0,0 +1,43 @@
+package config
+
+import (
+	"context"
+
+	"github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/fsnotify/fsnotify"
+)
+
+type HookActionType string
+
+const (
+	SystemdHookActionType    = "Systemd"
```

Comment:
this is optional as we can use executable but I think its worth it as systemd is so common and will be easy to support Ign systemd spec